s3tests_boto3/functional/test_s3.py

```diff
@@ -8607,6 +8607,8 @@ def _test_encryption_sse_customer_write(file_size):
         'x-amz-server-side-encryption-customer-key': 'pO3upElrwuEXSoFwCfnZPdSsmt/xWeFa0N9KgDijwVs=',
         'x-amz-server-side-encryption-customer-key-md5': 'DWygnHRtgiJ77HCm+1rvHw=='
     }
+    if get_config_is_secure() == False:
+        sse_client_headers['X-Forwarded-Proto'] = 'https'
```

Comment:
this shouldn't be necessary, and servers probably shouldn't accept this header from clients without additional configuration. if you're just testing against ceph rgw, you can set rgw_crypt_require_ssl=false to run these

Reply:
In my tests, if the `rgw_crypt_require_ssl=false` is set, the ceph rgw will require the header in order to accept the request.

Reply:
that config option has always worked in our automated testing, ex. https://github.com/ceph/ceph/blob/04416f4/qa/suites/rgw/crypt/3-rgw/rgw.yaml#L7